lib.php

```diff
@@ -154,13 +154,19 @@ public static function arraysyntax_tosql($arraysyntax) { // TODO : protected
                 $sub_sql = self::arraysyntax_tosql($sub_arraysyntax);
                 $select .= ' '.$sub_sql['select'].' ';
                 $where .= ' ( '.$sub_sql['where'].' ) ';
+                $params = array_merge($params,$sub_sql['params']);
             }
             else {
                 if ($customkey = array_search($rule->param, $customuserfields, true)) {
                     // custom user field actually exists
                     $join_id++;
-                    $select .= ' RIGHT JOIN '.$CFG->prefix.'user_info_data d'.$join_id.' ON d'.$join_id.'.userid = u.id';
-                    $where .= ' (d'.$join_id.'.fieldid = '.$customkey.' AND ( d'.$join_id.'.data = \''.$rule->value.'\' OR d'.$join_id.'.data LIKE \'%;'.$rule->value.'\' OR d'.$join_id.'.data LIKE \''.$rule->value.';%\' OR d'.$join_id.'.data LIKE \'%;'.$rule->value.';%\' ))';
+                    global $DB;
+                    $data = 'd'.$join_id.'.data';
+                    $select .= ' RIGHT JOIN {user_info_data} d'.$join_id.' ON d'.$join_id.'.userid = u.id';
+                    $where .= ' (d'.$join_id.'.fieldid = ? AND ('.$DB->sql_compare_text($data).' = '.$DB->sql_compare_text('?').
+                            ' OR '.$DB->sql_like($DB->sql_compare_text($data),'?').' OR '
```

Comment:
Trivial: missing blank space after "`,`" when using multiple arguments: here and in the other places in this commit (546b5cf118832b2f17b3459ddeeddc819f6098f8).